systems/console-bff/gateway/tests/invitation.test.ts

```diff
@@ -0,0 +1,332 @@
+import { logger } from "../../common/logger";
+import { getBaseURL, parseGatewayHeaders } from "../../common/utils";
+import { Context } from "../../invitation/context";
+import InvitationApi from "../../invitation/datasource/invitation_api";
+import { CreateInvitationResolver } from "../../invitation/resolver/createInvitation";
+import { DeleteInvitationResolver } from "../../invitation/resolver/deleteInvitation";
+import { GetInvitationResolver } from "../../invitation/resolver/getInvitation";
+import { GetInVitationsByOrgResolver } from "../../invitation/resolver/getInvitationByOrg";
+import { GetInvitationsResolver } from "../../invitation/resolver/getInvitations";
+import { UpdateInvitationResolver } from "../../invitation/resolver/updateInvitation";
+
+const token = process.env.TOKEN;
+const headers = {
+  cookie: "ukama_session=random-session",
+  token: token,
+};
+
+const parsedHeaders = parseGatewayHeaders(headers);
+const { orgName } = parsedHeaders;
+
+const createSchema = async () => {
+  return await buildSchema({
+    resolvers: [
+      CreateInvitationResolver,
+      DeleteInvitationResolver,
+      GetInvitationResolver,
+      GetInvitationsResolver,
+      GetInVitationsByOrgResolver,
+      UpdateInvitationResolver,
+    ],
+    validate: true,
+  });
+};
+
+const redisClient = createClient().on("error", error => {
+  logger.error(
+    `Error creating redis for ${SUB_GRAPHS.invitation.name} service, Error: ${error}`
+  );
+});
+
+const invitationAPi = new InvitationApi();
+let invitationId = "";
+const invitationData = {
+  email: "test-email@mail.com",
+  name: "testUser2",
+  role: "ROLE_VENDOR",
+};
+
+const startServer = async () => {
+  const schema = await createSchema();
+  const server = new ApolloServer<Context>({
+    schema,
+  });
+  await server.start();
+  return server;
+};
+
+describe("Invitation API integration test", () => {
+  let server: ApolloServer<Context>;
+
+  beforeAll(async () => {
+    server = await startServer();
+  });
+  afterAll(async () => {
+    await server.stop();
+  });
+  it("should create an invitation", async () => {
+    const CREATE_INVITATION = `mutation CreateInvitation($data: CreateInvitationInputDto!) {
+    createInvitation(data: $data) {
+      email
+      expireAt
+      id
+      name
+      role
+      link
+      userId
+      status
+    }
+  }`;
+    const res = await server.executeOperation(
+      {
+        query: CREATE_INVITATION,
+        variables: {
+          data: invitationData,
+        },
+      },
+      {
+        contextValue: await (async () => {
+          const baseURL = await getBaseURL(
+            SUB_GRAPHS.invitation.name,
+            orgName,
+            redisClient.isOpen ? redisClient : null
+          );
+          return {
+            dataSources: {
+              dataSource: invitationAPi,
+            },
+            baseURL: baseURL.message,
+            headers: parsedHeaders,
+          };
+        })(),
```

Comment:
Just noticed this Context Value code is duplicated multiple time. We can define this in `beforeAll` and assign its value to a variable then use it everywhere. It'll reduce lot of code duplication.